app/main.py

```diff
@@ -57,9 +79,71 @@ async def favicon():
 def get_language_code(language: Language | str):
     return type(language) is str and language[:2] or language.value
 
+if PROXY_PANEL_THROUGH_FASTAPI:
+    # Forward the websockets through a bridge
+    from fastapi import WebSocket
+    import websockets
+    import asyncio
+    from starlette.websockets import WebSocketDisconnect
+    from websockets.exceptions import ConnectionClosedOK
+
+    @app.websocket("/panel/{plot_key}/ws")
+    async def websocket_endpoint(ws_client: WebSocket, plot_key: str):
+        await ws_client.accept()
+        subprotocols = ws_client.headers['sec-websocket-protocol'].split(', ')
+        uri = f"ws://{SERVER_ADDRESS}:{PANEL_PORT}/{plot_key}/ws"
+        async with websockets.connect(uri, subprotocols=subprotocols) as ws_server:
+
+            async def listen_to_client():
+                try:
+                    while True:
+                        data = await ws_client.receive_text()
+                        await ws_server.send(data)
+                except WebSocketDisconnect:
+                    await ws_server.close()
+
+            async def listen_to_server():
+                try:
+                    while True:
+                        data = await ws_server.recv()
+                        await ws_client.send_text(data)
+                except ConnectionClosedOK:
+                    pass
+
+            await asyncio.gather(listen_to_client(), listen_to_server())
+
+    # Also stream/serve other panel specific content, especifically the autoload.js
+    from starlette.requests import Request
```

Comment:
These imports should go up as well, at least with the ones inside this same condition

Reply:
yep